### ISSUE TITLE
limited Options available on DR-HAF001S

### ISSUE THREAD

**Issue by ninlee** [JeffSteinbok/hass-dreo#49]
after installation of this integration i can add my DR-HAF001S fan but only have 3 enteries available to me.

Horizontally Oscillating
Panel Sound
Temperature

cant turn on or off cant change speed... can you help

**Comment by JeffSteinbok** [comment]
Please send me the diagnostics from the integration.
thx

**Comment by ninlee** [comment]
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2023.8.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.11.4",
    "docker": true,
    "arch": "x86_64",
    "timezone": "America/Port_of_Spain",
    "os_name": "Linux",
    "os_version": "6.1.45",
    "supervisor": "2023.08.1",
    "host_os": "Home Assistant OS 10.5",
    "docker_version": "23.0.6",
    "chassis": "vm",
    "run_as_root": true
  },
  "custom_components": {
    "xiaomi_cloud_map_extractor": {
      "version": "v2.2.0",
      "requirements": [
        "pillow",
        "pybase64",
        "python-miio",
        "requests",
        "pycryptodome"
      ]
    },
    "hacs": {
      "version": "1.32.1",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "hass_agent": {
      "version": "2022.11.9",
      "requirements": []
    },
    "alexa_media": {
      "version": "4.6.5",
      "requirements": [
        "alexapy==1.26.8",
        "packaging>=20.3",
        "wrapt>=1.12.1"
      ]
    },
    "pfsense": {
      "version": "0.1.0",
      "requirements": [
        "mac-vendor-lookup>=0.1.11"
      ]
    },
    "dwains_dashboard": {
      "version": "3.4.3",
      "requirements": []
    },
    "dreo": {
      "version": "0.0.2",
      "requirements": [
        "websockets"
      ]
    },
    "hass_agent_notifier": {
      "version": "2022.10.25.1",
      "requirements": []
    },
    "hass_agent_mediaplayer": {
      "version": "2022.5.6",
      "requirements": []
    },
    "smartir": {
      "version": "1.17.7",
      "requirements": [
        "aiofiles>=0.6.0"
      ]
    },
    "nodered": {
      "version": "2.2.0",
      "requirements": []
    },
    "browser_mod": {
      "version": "2.3.0",
      "requirements": []
    },
    "sonoff": {
      "version": "3.5.2",
      "requirements": [
        "pycryptodome>=3.6.6"
      ]
    },
    "frigate": {
      "version": "4.0.0",
      "requirements": [
        "pytz==2022.7"
      ]
    },
    "watchman": {
      "version": "0.5.1",
      "requirements": [
        "prettytable==3.0.0"
      ]
    },
    "reolink_dev": {
      "version": "0.64",
      "requirements": [
        "reolink==0.0.64",
        "aiosmtpd>=1.4.2"
      ]
    },
    "scheduler": {
      "version": "v0.0.0",
      "requirements": []
    },
    "alarmo": {
      "version": "v1.9.10",
      "requirements": []
    }
  },
  "integration_manifest": {
    "domain": "dreo",
    "name": "Dreo",
    "codeowners": [
      "@jeffsteinbok"
    ],
    "config_flow": true,
    "dependencies": [],
    "documentation": "https://github.com/jeffsteinbok/hass-dreo/blob/master/README.md",
    "iot_class": "cloud_push",
    "issue_tracker": "https://github.com/jeffsteinbok/hass-dreo/issues",
    "loggers": [
      "dreo",
      "pydreo"
    ],
    "requirements": [
      "websockets"
    ],
    "version": "0.0.2",
    "is_built_in": false
  },
  "data": {
    "dreo": {
      "fan_count": 1,
      "raw_devicelist": {
        "code": 0,
        "msg": "OK",
        "data": {
          "currentPage": 1,
          "pageSize": 100,
          "totalNum": 1,
          "totalPage": 1,
          "list": [
            {
              "deviceId": "1693759130586640385",
              "sn": "1572127056357535746-2e072b3b8590090c:001:0000000000s",
              "brand": "Dreo",
              "model": "DR-HAF001S",
              "productId": "1572127056357535746",
              "productName": "Air Circulator",
              "deviceName": "Air Circulator",
              "shared": false,
              "series": null,
              "seriesName": "CF511S",
              "controlsConf": {
                "template": "DR-HAF001S",
                "lottie": {
                  "key": "poweron",
                  "frames": [
                    {
                      "value": 0,
                      "frame": [
                        0
                      ]
                    },
                    {
                      "value": 1,
                      "frame": [
                        2
                      ]
                    }
                  ]
                },
                "cards": [
                  {
                    "type": 2,
                    "title": "device_control_temp",
                    "icon": "",
                    "image": "",
                    "url": "",
                    "show": true
                  },
                  {
                    "type": 6,
                    "title": "device_settings_title",
                    "icon": "ic_setting",
                    "image": "",
                    "url": "dreo://nav/device/setting?deviceSn=${sn}",
                    "show": true,
                    "key": "setting"
                  }
                ],
                "preference": [
                  {
                    "id": "200",
                    "type": "Panel Sound",
                    "title": "device_control_panelsound",
                    "image": "ic_mute",
                    "reverse": true,
                    "cmd": "muteon"
                  },
                  {
                    "id": "210",
                    "type": "Display Auto Off",
                    "title": "device_fans_mode_auto_display",
                    "image": "ic_display",
                    "reverse": true,
                    "cmd": "ledkepton"
                  },
                  {
                    "id": "220",
                    "type": "Child Lock",
                    "title": "device_control_childlock",
                    "image": "ic_child_lock",
                    "reverse": false,
                    "cmd": "childlockon"
                  },
                  {
                    "id": "240",
                    "type": "Temperature Unit",
                    "title": "device_control_temp_unit",
                    "image": "ic_temp_unit"
                  }
                ],
                "control": [
                  {
                    "id": "100",
                    "type": "Mode",
                    "title": "device_mode",
                    "items": [
                      {
                        "text": "device_fans_mode_straight",
                        "textColors": [
                          "#D5D6D7",
                          "#D5D6D7"
                        ],
                        "image": "ic_normal_wind",
                        "imageColors": [
                          "#D5D6D7",
                          "#25D7E4"
                        ],
                        "cmd": "mode",
                        "value": 1
                      },
                      {
                        "text": "device_fans_mode_natural",
                        "textColors": [
                          "#D5D6D7",
                          "#D5D6D7"
                        ],
                        "image": "ic_natural_wind",
                        "imageColors": [
                          "#D5D6D7",
                          "#2CDD96"
                        ],
                        "cmd": "mode",
                        "value": 2
                      },
                      {
                        "text": "device_control_mode_sleep",
                        "textColors": [
                          "#D5D6D7",
                          "#D5D6D7"
                        ],
                        "image": "ic_sleep_wind",
                        "imageColors": [
                          "#D5D6D7",
                          "#6249DF"
                        ],
                        "cmd": "mode",
                        "value": 3
                      },
                      {
                        "text": "device_control_mode_auto",
                        "textColors": [
                          "#D5D6D7",
                          "#D5D6D7"
                        ],
                        "image": "ic_auto_wind",
                        "imageColors": [
                          "#D5D6D7",
                          "#0A80F5"
                        ],
                        "cmd": "mode",
                        "value": 4
                      }
                    ]
                  },
                  {
                    "id": "110",
                    "type": "Speed",
                    "title": "device_control_speed",
                    "items": [
                      {
                        "text": "1",
                        "cmd": "windlevel",
                        "value": 1
                      },
                      {
                        "text": "4",
                        "cmd": "windlevel",
                        "value": 4
                      }
                    ]
                  },
                  {
                    "id": "130",
                    "type": "DPad",
                    "step": 5
                  },
                  {
                    "id": "120",
                    "type": "Oscillation",
                    "title": "device_control_oscillation",
                    "items": [],
                    "cmd": "hoscon"
                  }
                ],
                "category": "Air Circulators",
                "version": {
                  "minControlVer": "2.0.7",
                  "minPairingVer": "2.0.7"
                }
              },
              "mainConf": {
                "isSmart": true,
                "isWifi": true,
                "isBluetooth": true,
                "isVoiceControl": true
              },
              "resourcesConf": {
                "imageSmallSrc": "https://resources.dreo-cloud.com/app/202302/1562f474986ecd484f870a6b1874406b82.png",
                "imageFullSrc": "https://resources.dreo-cloud.com/app/202307/25e0425cca23dc4b9692de9486b7f23ec0.zip"
              },
              "servicesConf": [
                {
                  "key": "user_manual",
                  "value": "https://resources.dreo-cloud.com/app/202302/152e30ca6041664eb1802ac43c22ec476a.pdf"
                }
              ]
            }
          ]
        }
      }
    },
    "devices": {
      "fans": [
        {
          "_name": "Air Circulator",
          "_device_id": "1693759130586640385",
          "_sn": "**REDACTED**",
          "_model": "DR-HAF001S",
          "_dreo": {
            "__type": "<class 'custom_components.dreo.pydreo.PyDreo'>",
            "repr": "<custom_components.dreo.pydreo.PyDreo object at 0x7fdde5c8bb10>"
          },
          "_is_on": false,
          "_feature_key_names": {},
          "raw_state": {
            "code": 0,
            "msg": "OK",
            "data": {
              "mixed": {
                "mcu_hardware_model": {
                  "state": "SC95F8613B",
                  "timestamp": 1692756400
                },
                "wifi_ssid": "**REDACTED**",
                "windlevel": {
                  "state": 3,
                  "timestamp": 1692756400
                },
                "wifi_rssi": {
                  "state": -10,
                  "timestamp": 1692756400
                },
                "poweron": {
                  "state": false,
                  "timestamp": 1692756400
                },
                "tempunit": {
                  "state": 1,
                  "timestamp": 1692756400
                },
                "timeron": {
                  "state": {
                    "du": 0,
                    "ts": 1692756399
                  },
                  "timestamp": null
                },
                "module_firmware_version": {
                  "state": "1.2.12",
                  "timestamp": 1692756400
                },
                "mode": {
                  "state": 6,
                  "timestamp": 1692756400
                },
                "mcuon": {
                  "state": true,
                  "timestamp": 1692756400
                },
                "connected": {
                  "state": true,
                  "timestamp": 1692756400
                },
                "timeroff": {
                  "state": {
                    "du": 0,
                    "ts": 1692756399
                  },
                  "timestamp": null
                },
                "network_latency": {
                  "state": 152,
                  "timestamp": 1692756400
                },
                "module_hardware_model": {
                  "state": "PAI-051",
                  "timestamp": 1692756400
                },
                "mcu_firmware_version": {
                  "state": "1.0.17",
                  "timestamp": 1692756400
                },
                "customconf": {
                  "state": "temp:1111122222233333344444",
                  "timestamp": 1692756400
                },
                "ledkepton": {
                  "state": false,
                  "timestamp": 1692756400
                },
                "temperature": {
                  "state": 92,
                  "timestamp": 1692812917
                },
                "module_hardware_mac": "**REDACTED**",
                "childlockon": {
                  "state": false,
                  "timestamp": 1692756400
                },
                "muteon": {
                  "state": false,
                  "timestamp": 1692756400
                },
                "hoscon": {
                  "state": true,
                  "timestamp": 1692756400
                }
              },
              "sn": "**REDACTED**",
              "productId": "1572127056357535746",
              "region": "us-east-2"
            }
          },
          "_attr_cbs": [
            {
              "__type": "<class 'function'>",
              "repr": "<function DreoBaseDeviceHA.async_added_to_hass.<locals>.update_state at 0x7fdddb5b6480>"
            },
            {
              "__type": "<class 'function'>",
              "repr": "<function DreoBaseDeviceHA.async_added_to_hass.<locals>.update_state at 0x7fdddd5a94e0>"
            },
            {
              "__type": "<class 'function'>",
              "repr": "<function DreoBaseDeviceHA.async_added_to_hass.<locals>.update_state at 0x7fdddd5a9620>"
            }
          ],
          "_lock": {
            "__type": "<class '_thread.lock'>",
            "repr": "<unlocked _thread.lock object at 0x7fdde5419940>"
          },
          "_fan_definition": {
            "__type": "<class 'custom_components.dreo.pydreo.models.DreoFanDetails'>",
            "repr": "DreoFanDetails(preset_modes=['normal', 'natural', 'sleep', 'auto', 'turbo'], speed_range=(1, 4))"
          },
          "_fan_speed": 3,
          "_wind_type": null,
          "_wind_mode": 6,
          "_shakehorizon": null,
          "_osc_mode": null,
          "_horizontally_oscillating": true,
          "_vertically_oscillating": null,
          "_temperature": 92,
          "_led_always_on": null,
          "_voice_on": null,
          "_light_sensor_on": null,
          "_mute_on": false,
          "_fixed_conf": null
        }
      ]
    }
  }
}

**Comment by JeffSteinbok** [comment]
Ok - What do you see in HA?  I see that the integration sees fan speed and power state.

**Comment by ninlee** [comment]
![dreo](https://github.com/JeffSteinbok/hass-dreo/assets/73006242/20dd558a-9f8d-4900-826b-963b364380c3)


just :
Horizontally Oscillating
Panel Sound
Temperature

**Comment by JeffSteinbok** [comment]
No fan entry at all?
Never seen that at all.  Any chance it's disabled or something?
If you could, turn on debug logging and restart HA and let me know if it has any info, or just attach the logs.  I've never seen it NOT have the fan entry at all.

**Comment by ninlee** [comment]
how exactly am i enabling the logs?

**Comment by ninlee** [comment]
add the lines to the configuration.ymal? 

if so i tried.

**Comment by JeffSteinbok** [comment]
And then pull the logs from HomeAssistant.
I've never seen the case where the sensors/switches show up, but not the fan itself

**Comment by daschu117** [comment]
I similarly don't have a fan entity for a DR-HAF003S when first setting up this integration. Only see the following:
![image](https://github.com/JeffSteinbok/hass-dreo/assets/576080/8ef3f709-c04f-4406-9cd9-1c89a467cf91)

Diagnostics:
```
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2023.8.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.11.4",
    "docker": true,
    "arch": "aarch64",
    "timezone": "America/Los_Angeles",
    "os_name": "Linux",
    "os_version": "6.1.21-v8",
    "supervisor": "2023.08.1",
    "host_os": "Home Assistant OS 10.5",
    "docker_version": "23.0.6",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "mila": {
      "version": "0.4.3",
      "requirements": [
        "milasdk==0.4.3",
        "python-benedict==0.24.3",
        "geopy==2.2.0",
        "python-aqi==0.6.1"
      ]
    },
    "uptime_kuma": {
      "version": "2.1.0",
      "requirements": [
        "pyuptimekuma-hass"
      ]
    },
    "midea_ac_lan": {
      "version": "v0.3.18",
      "requirements": []
    },
    "hacs": {
      "version": "1.32.1",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "hass_agent": {
      "version": "2022.11.9",
      "requirements": []
    },
    "steam_wishlist": {
      "version": "2.2.3",
      "requirements": []
    },
    "emporia_vue": {
      "version": "0.8.3",
      "requirements": [
        "pyemvue==0.17.1"
      ]
    },
    "dreo": {
      "version": "0.0.2",
      "requirements": [
        "websockets"
      ]
    }
  },
  "integration_manifest": {
    "domain": "dreo",
    "name": "Dreo",
    "codeowners": [
      "@jeffsteinbok"
    ],
    "config_flow": true,
    "dependencies": [],
    "documentation": "https://github.com/jeffsteinbok/hass-dreo/blob/master/README.md",
    "iot_class": "cloud_push",
    "issue_tracker": "https://github.com/jeffsteinbok/hass-dreo/issues",
    "loggers": [
      "dreo",
      "pydreo"
    ],
    "requirements": [
      "websockets"
    ],
    "version": "0.0.2",
    "is_built_in": false
  },
  "data": {
    "dreo": {
      "fan_count": 1,
      "raw_devicelist": {
        "code": 0,
        "msg": "OK",
        "data": {
          "currentPage": 1,
          "pageSize": 100,
          "totalNum": 1,
          "totalPage": 1,
          "list": [
            {
              "deviceId": "1694575334046830594",
              "sn": "1562320538242166785-fd8b9ebf8bb7f8f7:001:0000000000l",
              "brand": "Dreo",
              "model": "DR-HAF003S",
              "productId": "1562320538242166785",
              "productName": "Air Circulator",
              "deviceName": "Air Circulator",
              "shared": false,
              "series": null,
              "seriesName": "PolyFan 513S",
              "controlsConf": {
                "template": "DR-HAF003S",
                "lottie": {
                  "key": "poweron",
                  "frames": [
                    {
                      "value": 0,
                      "frame": [
                        0
                      ]
                    },
                    {
                      "value": 1,
                      "frame": [
                        2
                      ]
                    }
                  ]
                },
                "cards": [
                  {
                    "type": 2,
                    "title": "device_control_temp",
                    "icon": "",
                    "image": "",
                    "url": "",
                    "show": true
                  },
                  {
                    "type": 6,
                    "title": "device_settings_title",
                    "icon": "ic_setting",
                    "image": "",
                    "url": "dreo://nav/device/setting?deviceSn=${sn}",
                    "show": true,
                    "key": "setting"
                  }
                ],
                "preference": [
                  {
                    "id": "200",
                    "type": "Panel Sound",
                    "title": "device_control_panelsound",
                    "image": "ic_mute",
                    "reverse": true,
                    "cmd": "muteon"
                  },
                  {
                    "id": "210",
                    "type": "Display Auto Off",
                    "title": "device_fans_mode_auto_display",
                    "image": "ic_display",
                    "reverse": true,
                    "cmd": "lighton"
                  },
                  {
                    "id": "220",
                    "type": "Child Lock",
                    "title": "device_control_childlock",
                    "image": "ic_child_lock",
                    "reverse": false,
                    "cmd": "childlockon"
                  },
                  {
                    "id": "230",
                    "type": "Oscillation Calibration",
                    "title": "device_control_calibration",
                    "image": "ic_osc_cal"
                  },
                  {
                    "id": "240",
                    "type": "Temperature Unit",
                    "title": "device_control_temp_unit",
                    "image": "ic_temp_unit"
                  }
                ],
                "control": [
                  {
                    "id": "100",
                    "type": "Mode",
                    "title": "device_mode",
                    "items": [
                      {
                        "text": "device_fans_mode_straight",
                        "textColors": [
                          "#D5D6D7",
                          "#D5D6D7"
                        ],
                        "image": "ic_normal_wind",
                        "imageColors": [
                          "#D5D6D7",
                          "#25D7E4"
                        ],
                        "cmd": "mode",
                        "value": 1
                      },
                      {
                        "text": "device_fans_mode_natural",
                        "textColors": [
                          "#D5D6D7",
                          "#D5D6D7"
                        ],
                        "image": "ic_natural_wind",
                        "imageColors": [
                          "#D5D6D7",
                          "#2CDD96"
                        ],
                        "cmd": "mode",
                        "value": 2
                      },
                      {
                        "text": "device_control_mode_sleep",
                        "textColors": [
                          "#D5D6D7",
                          "#D5D6D7"
                        ],
                        "image": "ic_sleep_wind",
                        "imageColors": [
                          "#D5D6D7",
                          "#6249DF"
                        ],
                        "cmd": "mode",
                        "value": 3
                      },
                      {
                        "text": "device_control_mode_auto",
                        "textColors": [
                          "#D5D6D7",
                          "#D5D6D7"
                        ],
                        "image": "ic_auto_wind",
                        "imageColors": [
                          "#D5D6D7",
                          "#0A80F5"
                        ],
                        "cmd": "mode",
                        "value": 4
                      }
                    ]
                  },
                  {
                    "id": "110",
                    "type": "Speed",
                    "title": "device_control_speed",
                    "items": [
                      {
                        "text": "1",
                        "cmd": "windlevel",
                        "value": 1
                      },
                      {
                        "text": "8",
                        "cmd": "windlevel",
                        "value": 8
                      }
                    ]
                  },
                  {
                    "id": "130",
                    "type": "DPad",
                    "step": 5
                  }
                ],
                "category": "Air Circulators",
                "version": {
                  "minControlVer": "2.0.7",
                  "minPairingVer": "2.0.7"
                }
              },
              "mainConf": {
                "isSmart": true,
                "isWifi": true,
                "isBluetooth": true,
                "isVoiceControl": true
              },
              "resourcesConf": {
                "imageSmallSrc": "https://resources.dreo-cloud.com/app/202302/15a50a18703f4240f3b4f91072406051ac.png",
                "imageFullSrc": "https://resources.dreo-cloud.com/app/202308/180b23cca29dc44433a89deb0f3f1b3140.zip"
              },
              "servicesConf": [
                {
                  "key": "user_manual",
                  "value": "https://resources.dreo-cloud.com/app/202302/1532125196c1c54605903cb2e5880415c3.pdf"
                }
              ]
            }
          ]
        }
      }
    },
    "devices": {
      "fans": [
        {
          "_name": "Air Circulator",
          "_device_id": "1694575334046830594",
          "_sn": "**REDACTED**",
          "_model": "DR-HAF003S",
          "_dreo": {
            "__type": "<class 'custom_components.dreo.pydreo.PyDreo'>",
            "repr": "<custom_components.dreo.pydreo.PyDreo object at 0x7f7c41e910>"
          },
          "_is_on": true,
          "_feature_key_names": {},
          "raw_state": {
            "code": 0,
            "msg": "OK",
            "data": {
              "mixed": {
                "wifi_rssi": {
                  "state": -59,
                  "timestamp": 1692859776
                },
                "poweron": {
                  "state": true,
                  "timestamp": 1692859776
                },
                "voscon": {
                  "state": false,
                  "timestamp": 1692859776
                },
                "hoscangle": {
                  "state": 30,
                  "timestamp": 1692859776
                },
                "timeron": {
                  "state": {
                    "du": 0,
                    "ts": 851
                  },
                  "timestamp": null
                },
                "mode": {
                  "state": 6,
                  "timestamp": 1692859776
                },
                "mcuon": {
                  "state": true,
                  "timestamp": 1692853224
                },
                "network_latency": {
                  "state": 154,
                  "timestamp": 1692859776
                },
                "module_hardware_model": {
                  "state": "PAI-051",
                  "timestamp": 1692859776
                },
                "mcu_firmware_version": {
                  "state": "0.1.1",
                  "timestamp": 1692853224
                },
                "customconf": {
                  "state": "temp:12233556688888888888888",
                  "timestamp": 1692859776
                },
                "temperature": {
                  "state": 73,
                  "timestamp": 1692861792
                },
                "module_hardware_mac": "**REDACTED**",
                "hoscreset": {
                  "state": false,
                  "timestamp": 1692859776
                },
                "alignon": {
                  "state": true,
                  "timestamp": 1692859776
                },
                "muteon": {
                  "state": true,
                  "timestamp": 1692859776
                },
                "lighton": {
                  "state": false,
                  "timestamp": 1692859776
                },
                "mcu_hardware_model": {
                  "state": "HAF003S",
                  "timestamp": 1692853224
                },
                "wifi_ssid": "**REDACTED**",
                "windlevel": {
                  "state": 5,
                  "timestamp": 1692859776
                },
                "voscreset": {
                  "state": false,
                  "timestamp": 1692859776
                },
                "wrong": {
                  "state": 0,
                  "timestamp": 1692859776
                },
                "module_firmware_version": {
                  "state": "1.2.19",
                  "timestamp": 1692859776
                },
                "connected": {
                  "state": true,
                  "timestamp": 1692859776
                },
                "timeroff": {
                  "state": {
                    "du": 0,
                    "ts": 851
                  },
                  "timestamp": null
                },
                "childlockon": {
                  "state": false,
                  "timestamp": 1692859776
                },
                "voscangle": {
                  "state": 0,
                  "timestamp": 1692859776
                },
                "hoscon": {
                  "state": true,
                  "timestamp": 1692859776
                }
              },
              "sn": "**REDACTED**",
              "productId": "1562320538242166785",
              "region": "us-east-2"
            }
          },
          "_attr_cbs": [
            {
              "__type": "<class 'function'>",
              "repr": "<function DreoBaseDeviceHA.async_added_to_hass.<locals>.update_state at 0x7f7c169580>"
            },
            {
              "__type": "<class 'function'>",
              "repr": "<function DreoBaseDeviceHA.async_added_to_hass.<locals>.update_state at 0x7f7bc7ce00>"
            },
            {
              "__type": "<class 'function'>",
              "repr": "<function DreoBaseDeviceHA.async_added_to_hass.<locals>.update_state at 0x7f7bc7ccc0>"
            },
            {
              "__type": "<class 'function'>",
              "repr": "<function DreoBaseDeviceHA.async_added_to_hass.<locals>.update_state at 0x7f7bc7ca40>"
            }
          ],
          "_lock": {
            "__type": "<class '_thread.lock'>",
            "repr": "<unlocked _thread.lock object at 0x7f82d66d00>"
          },
          "_fan_definition": {
            "__type": "<class 'custom_components.dreo.pydreo.models.DreoFanDetails'>",
            "repr": "DreoFanDetails(preset_modes=['normal', 'natural', 'sleep', 'auto', 'turbo'], speed_range=(1, 8))"
          },
          "_fan_speed": 5,
          "_wind_type": null,
          "_wind_mode": 6,
          "_shakehorizon": null,
          "_osc_mode": null,
          "_horizontally_oscillating": true,
          "_vertically_oscillating": false,
          "_temperature": 73,
          "_led_always_on": null,
          "_voice_on": null,
          "_light_sensor_on": null,
          "_mute_on": true,
          "_fixed_conf": null
        }
      ]
    }
  }
}
```

Debug logging:
```
2023-08-24 01:17:22.842 ERROR (MainThread) [homeassistant.components.fan] Error adding entities for domain fan with platform dreo
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 660, in _async_add_entity
    capabilities=entity.capability_attributes,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/fan/__init__.py", line 334, in capability_attributes
    self.supported_features & FanEntityFeature.SET_SPEED
    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/dreo/fan.py", line 93, in supported_features
    if (self.device.preset_mode is not None):
        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/dreo/pydreo/pydreofan.py", line 114, in preset_mode
    return self.preset_modes[mode - 1]
           ~~~~~~~~~~~~~~~~~^^^^^^^^^^
IndexError: list index out of range
2023-08-24 01:17:22.902 ERROR (MainThread) [homeassistant.components.fan] Error while setting up dreo platform for fan
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 370, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 660, in _async_add_entity
    capabilities=entity.capability_attributes,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/fan/__init__.py", line 334, in capability_attributes
    self.supported_features & FanEntityFeature.SET_SPEED
    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/dreo/fan.py", line 93, in supported_features
    if (self.device.preset_mode is not None):
        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/dreo/pydreo/pydreofan.py", line 114, in preset_mode
    return self.preset_modes[mode - 1]
           ~~~~~~~~~~~~~~~~~^^^^^^^^^^
IndexError: list index out of range
2023-08-24 01:17:43.591 DEBUG (MainThread) [dreo] Options Flow Step Init
2023-08-24 01:17:43.592 DEBUG (MainThread) [dreo] auto_reconnect not set, setting it to True
2023-08-24 01:17:45.349 DEBUG (MainThread) [dreo] Options Flow Step Init
2023-08-24 01:17:45.349 DEBUG (MainThread) [dreo] UserInput is not none
2023-08-24 01:17:47.156 DEBUG (MainThread) [dreo] Options Flow Step Init
2023-08-24 01:17:48.313 DEBUG (MainThread) [dreo] Options Flow Step Init
2023-08-24 01:17:48.313 DEBUG (MainThread) [dreo] UserInput is not none
```

**Comment by JeffSteinbok** [comment]
First thing first, thank you for trying the integration and giving feedback.
Secondly, I'm sorry that you're seeing these bugs. I haven't seen this particular one.

Ok...

- Re the logs:  it should be "debug", not "verbose".  I'll fix the README.
- The logs above show me exactly what's wrong, and I expect I can fix today/tomorrow.

Thank you.  Stay tuned.

**Comment by JeffSteinbok** [comment]
@daschu117 - Your fan is returning preset mode 6, but i dont' see 6 in the list of supported modes.  What mode is it set to in the app?

**Comment by daschu117** [comment]
@JeffSteinbok - I had the fan in the preset "Use with AC" listed under the "Custom" tab of modes. Based on what you said about it being in an unexpected mode, I removed my Dreo account from Home Assistant, changed the fan to Standard in the app, then re-added the Dreo account to the integration. Now I see a Fan entity of the device.

The fan is brand new to me and I bought it specifically because I had seen this integration, so if there's more I can do to help test I'm happy to mess around with it.

Thanks!

**Comment by ninlee** [comment]
> 

This seemed to have worked for me also thanks

**Comment by JeffSteinbok** [comment]
Excellent.
Please follow steps here:  https://github.com/JeffSteinbok/hass-dreo#debug-logs-for-new-fans 
That should give me some info for what the unexpected values are.

**Comment by JeffSteinbok** [comment]
Fixed in 0.6.1 - Opened #54 to track adding full set of modes.  @ninlee and @daschu117  - Please have a look.
